documentation/general/decouple-vs-and-net-sdk.md

```diff
@@ -0,0 +1,180 @@
+
+## Torn State Scenarios
+
+There are a number of scenarios where customers end up in a torn state. The most common is when customers have a CI setup with the following items:
+
+1. They use a task like [actions/setup-dotnet][setup-dotnet] to install the .NET SDK and use a flexible version format like `8.x`.
+2. They use `msbuild` to actually drive their build. This `msbuild` comes from the Visual Studio installed on the machine image.
+
+This means that CI systems are updated to the latest .NET SDK virtually as soon as we release them. However the version of Visual Studio is updated much later as CI images usually take several weeks to upgrade to a new Visual Studio version. This is a very common CI setup and means a significant number of our customers end up in a torn state for several weeks.
+
+Teams also get into this state when the Visual Studio used for developement is older than the .NET SDK they are using:
+
+1. Teams can get locked into older Visual Studio via org policy but freely update to newer .NET SDKs.
+2. Using a preview version of the .NET SDK. These inherently represent a torn SDK state because they almost never match the compiler in Visual Studio. This results in blockers for big teams like Bing from testing out our previews.
+3. Using a global.json that specifies a different version of the .NET SDK than the Visual Studio version.
+
+## Solution
+
+### MSBuild will use the .NET SDK tools for SDK projects
+
+When building .NET SDK projects, `msbuild` will load tasks from the .NET SDK instead of Visual Studio. Specifically it will load the Roslyn compiler task from the .NET SDK. That task will then function the same as it does when launched from `dotnet build`.
+
+To facilitate tasks from the .NET SDK launching .NET Core based processes, the `msbuild` host will [set the environment variable][dotnet-host-path] `DOTNET_HOST_PATH`. This will point to the same .NET Core host that the .NET SDK uses. This will allow tasks to launch .NET Core based processes with the same behavior as the .NET SDK would.
+
+This change will allow for `msbuild` to have a more consistent build experience with `dotnet build` for .NET SDK projects. It will have no impact on non-SDK projects as they will continue to use the compiler tasks that come from Visual Studio.
+
+## .NET SDK analyzers will load from Visual Studio
+
+Analyzers and generators which ship in the .NET SDK box will change to having a copy checked into Visual Studio. This will occur as part of the .NET SDK insertion process. When .NET SDK based projects are loaded at design time, the Visual Studio copy of the analyzer will be loaded.
+
+This means that the behavior of analyzers in Visual Studio design time will be independent of the .NET SDK used by the project. This increases the predictability of our product as we know at ship time what analyzer and generator experience our customers will be getting. It will no longer change as the underyling SDK does.
```

Comment:
```suggestion
This means that the behavior of analyzers in Visual Studio design time will be independent of the .NET SDK used by the project. This increases the predictability of our product as we know at ship time what analyzer and generator experience our customers will be getting. It will no longer change as the underlying SDK does.
```